src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/ILToLLVMImporter.cs

```diff
@@ -5100,7 +5100,7 @@ private void ImportStoreElement(TypeDesc elementType)
             StackEntry index = _stack.Pop();
             StackEntry arrayReference = _stack.Pop();
             var nullSafeElementType = elementType ?? GetWellKnownType(WellKnownType.Object);
-            LLVMValueRef elementAddress = GetElementAddress(index.ValueAsInt32(_builder, true), arrayReference.ValueAsType(LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0), _builder), nullSafeElementType);
+            LLVMValueRef elementAddress = GetElementAddress(index.ValueAsInt32(_builder, false), arrayReference.ValueAsType(LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0), _builder), nullSafeElementType);
```

Comment:
Zero-extending signed small types does not look correct, but then the values for which that would matter (negative ones) would also induce an IORE, which this code (seemingly) doesn't throw, and presumably we're not investing in fixing that, so it is fine to leave things be as they are.

Reply:
Yes, we are missing IORE.

Reply:
...which would have saved me hours of debugging, but heh ho.

Reply:
@jkotas Sorry to prod, but I can't build working packages locally for some reason at the moment.  If you can merge this that would be great, thanks